tuplex/test/core/IteratorTest.cc

```diff
@@ -0,0 +1,460 @@
+                "    return (b1, b2)";
+
+    auto v = c.parallelize({
+        Row(0)
+    }).map(UDF(func)).collectAsVector();
+
+    EXPECT_EQ(v.size(), 1);
+    EXPECT_EQ(v[0], Row("None", 0));
+}
+
+TEST_F(IteratorTest, CodegenTestNestedIteratorI) {
+    using namespace tuplex;
+    Context c(microTestOptions());
+
+    auto func = "def f(x):\n"
+                "    a = enumerate(iter(enumerate(iter([-1, -2, -3, -4]))))\n"
+                "    b = zip(a, 'abcd', enumerate(zip([1, 2], [3, 4])), zip(('A', 'B'), ('C', 'D')))\n"
+                "    c = enumerate(b, 10)\n"
+                "    d = iter(zip(iter(c), a))\n"
+                "    e1 = next(d)\n"
+                "    e2 = next(d)\n"
+                "    return (e1, e2)";
+
+    auto v = c.parallelize({
+        Row(0)
+    }).map(UDF(func)).collectAsVector();
+
+    EXPECT_EQ(v.size(), 1);
+    EXPECT_EQ(v[0].toPythonString(), "(((10,((0,(0,-1)),'a',(0,(1,3)),('A','C'))),(1,(1,-2))),((11,((2,(2,-3)),'b',(1,(2,4)),('B','D'))),(3,(3,-4))))");
+}
+
+TEST_F(IteratorTest, CodegenTestNestedIteratorII) {
+    using namespace tuplex;
+    Context c(microTestOptions());
+
+    auto func = "def f(x):\n"
+                "    a = [0, 1, 2, 3, 4, 5, 6]\n"
+                "    b = (10, 20, 30, 40, 50, 60)\n"
+                "    c = 'ABCDEFG'\n"
+                "    d = ('a', 'b', 'c', 'd', 'e', 'f')\n"
+                "    e = zip(a, b, c, d)\n"
+                "    f = zip(e, e, a, b)\n"
+                "    g = zip(f, e, c, d)\n"
+                "    h1 = next(g)\n"
+                "    h2 = next(g)\n"
+                "    return (h1, h2)";
+
+    auto v = c.parallelize({
+        Row(0)
+    }).map(UDF(func)).collectAsVector();
+
+    EXPECT_EQ(v.size(), 1);
+    EXPECT_EQ(v[0].toPythonString(), "((((0,10,'A','a'),(1,20,'B','b'),0,10),(2,30,'C','c'),'A','a'),(((3,40,'D','d'),(4,50,'E','e'),1,20),(5,60,'F','f'),'B','b'))");
+}
+
+TEST_F(IteratorTest, CodegenTestIfWithIteratorI) {
+    using namespace tuplex;
+    Context c(microTestOptions());
+
+    auto func = "def f(x):\n"
+                "    if x > 0:\n"
+                "        b = iter([1, 2])\n"
+                "        c = next(b)\n"
+                "        return c\n"
+                "    return x\n";
+
+    auto v = c.parallelize({
+        Row(10)
+    }).map(UDF(func)).collectAsVector();
+
+    EXPECT_EQ(v.size(), 1);
+    EXPECT_EQ(v[0], 1);
+}
+
+TEST_F(IteratorTest, CodegenTestIfWithIteratorII) {
+    using namespace tuplex;
+    Context c(microTestOptions());
+
+    auto func = "def f(x):\n"
+                "    a = iter(x)\n"
+                "    while True:\n"
+                "        if next(a) == 3:\n"
+                "            break\n"
+                "        else:\n"
+                "            next(a)\n"
+                "    return next(a)\n";
+
+    auto v = c.parallelize({
+        Row(List(1, 2, 3, 4))
+    }).map(UDF(func)).collectAsVector();
+
```

Comment:
Can we maybe add some test that use a bit of nested if logic?

E.g., 
```
if x > 0:
   next(x)
   y = next(x)
else:
  y = 20
```

Just to make sure using iterators within if-statements works properly.

Reply:
Added tests.